nav2_theta_star_planner/src/theta_star_planner.cpp

```diff
@@ -89,13 +89,27 @@ nav_msgs::msg::Path ThetaStarPlanner::createPlan(
 
   // Corner case of start and goal beeing on the same cell
   unsigned int mx_start, my_start, mx_goal, my_goal;
-  planner_->costmap_->worldToMap(start.pose.position.x, start.pose.position.y, mx_start, my_start);
-  planner_->costmap_->worldToMap(goal.pose.position.x, goal.pose.position.y, mx_goal, my_goal);
+  if (!planner_->costmap_->worldToMap(
+      start.pose.position.x, start.pose.position.y, mx_start, my_start))
+  {
+    throw nav2_core::StartOutsideMapBounds("Start outside map bounds");
+  }
+
+  if (!planner_->costmap_->worldToMap(
+      goal.pose.position.x, goal.pose.position.y, mx_goal, my_goal))
+  {
+    throw nav2_core::GoalOutsideMapBounds("Goal outside map bounds");
+  }
+
+  if (planner_->costmap_->getCost(mx_start, my_start) == nav2_costmap_2d::LETHAL_OBSTACLE) {
+    throw nav2_core::StartOccupied("Failed to create a unique pose path because of obstacles");
```

Comment:
Mention start/goal (depending which is being thrown) for logging in all of the `XYZOccupied` messages please